### PR TITLE
URL Shortener with database

### PR DESCRIPTION
**Pull request by amikhailau-ms**
Added solution for DatabaseHandler, that uses database instead of a map to store paths and URLs.

### CODE REVIEW

File: choose_your_own_adventure/adventure.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	"os"
+	"fmt"
+	"http/template"
+	"encoding/json"
+	"github.com/gopher-training/choose_your_own_adventure"
+
+)
+
+func main() {
+	fileJSON, err := os.Open("cyoa.json")
```
Comment (valadzko) [inline choose_your_own_adventure/adventure.go:13]: defer fileJSON.Close()

File: url_shortener_v2/urlWithDatabase.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	//yamlHandler, err := urlshort.YAMLHandler([]byte(yaml), mapHandler)
+	//if err != nil {
+	//	panic(err)
+	//}
+
+	//Database should be named after what you see in connStr
+	//and table should be named path_to_url.
+	//Values should be imported from csv file.
+	connStr := "dbname=url_shortener_database user=postgres password=postgres"
+	dbConnection, err := sql.Open("postgres", connStr)
```
Comment (valadzko) [inline url_shortener_v2/urlWithDatabase.go:39]: defer dbConnection.Close()

File: url_shortener_v2/urlshort/handler.go [hunk 3]
```diff
@@ -66,3 +68,23 @@ func parseYaml(yml []byte) (map[string]string, error) {
 	}
 	return parsedYaml, nil
 }
+
+//DatabaseHandler is the same as MapHandler, with exception that he reads from database
+func DatabaseHandler(db *sql.DB, fallback http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fmt.Println(r.URL.Path)
+		answer, err := (*db).Query("SELECT url FROM path_to_url WHERE path = $1;", r.URL.Path)
```
Comment (valadzko) [inline url_shortener_v2/urlshort/handler.go:76]: don't forget to close _answer_ to release connection from pool

**Review by valadzko — APPROVED** [review]

File: choose_your_own_adventure/adventure.go (hunk 1 above)
Comment (amikhailau-ms) [reply choose_your_own_adventure/adventure.go:13]: I'll update it later, in choose_your_adventure branch.

**Pull request merged**